File: 0287-find-the-duplicate-number/solution.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
       unordered_set<int> m;
       for(int num : nums){
        if(m.count(num)){
            return num;
        }
        m.insert(num);
       }
       return 0;
    }
};

File: 0287-find-the-duplicate-number/solution.java
class Solution {
    public int findDuplicate(int[] nums) {
        for(int i = 0 ;i<nums.length;i++){
            int val = Math.abs(nums[i]);
            if(nums[val] < 0){
                return val;
            }
            nums[val] = -nums[val];
        }
        return -1;
    }
}
